### ISSUE TITLE
[Bug] why we need the openai-key?

### ISSUE THREAD

**Issue by sytpb** [embedchain/embedchain#190]
Hi embedchain deploy on GPT4All ,why we need the openai-key ?  something about this I am not understand.

**Comment by aaishikdutta** [comment]
Noticed this for an edge case

I was working on using OpenSourceApp with InitConfig. I created a new parameter for initConfig in my forked repo named stream_response which is boolean and instantiated my OpenSourceApp like so:

```
init_config = InitConfig(stream_response=True)

app = OpenSourceApp(init_config)
```

This breaks saying need OpenAI API key. 

The problem here is InitConfig constructor has a condition that if `ef` is not passed to it, it will instantiate with the OpenAI embedding wrapper provided by chromadb, this executes before we are able to come to OpenSourceApp instantiation which checks for config existence and add `ef` as `all-MiniLM-L6-v2`. 

This is a bug @cachho, @taranjeet.

**Comment by cachho** [comment]
So this only happens when you instantiate an open source app with a config where `ef` is None.

**Comment by cachho** [comment]
The only way I see around this is by using the same architecture in App.__init__ that we use  in OpenSourceApp.__init__, where the default is not part of the config class, but instead set by the app class after initialization.

This might not be breakable by the end user, but I still think defaults belong in the config. Hmm, I don't like it but I see and acknowledge the issue.

**Comment by cachho** [comment]
proposed the fixes in my PR #205 - I mean: why not, i might not like the idea but it should fix the problem

**Comment by sytpb** [comment]
> 

init_config = InitConfig(stream_response=True)

app = OpenSourceApp(init_config)

Do this config don't need openai-key any more ?
